crates/iceberg/src/transaction/update_schema.rs

```diff
@@ -0,0 +1,246 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::collections::HashSet;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+
+use crate::error::{Error, ErrorKind, Result};
+use crate::spec::{Schema, SchemaRef, TableMetadataBuilder};
+use crate::table::Table;
+use crate::transaction::{ActionCommit, TransactionAction};
+use crate::{TableRequirement, TableUpdate};
+
+/// UpdateSchemaAction is a transaction action for performing schema evolution to the table.
+///
+/// TODO(hjiang): Currently only drop column is supported, need to implement a few more schema evolution operation, i.e., add columns, rename columns, change nullability, update default, etc.
+pub struct UpdateSchemaAction {
+    /// Schema update attributes.
+    case_sensitive: bool,
+    /// Current schema before update.
+    schema: SchemaRef,
+    /// Current field ids.
+    identifier_field_ids: HashSet<i32>,
+    /// Columns to drop on the table.
+    deletes: HashSet<i32>,
+}
+
+impl UpdateSchemaAction {
+    pub(crate) fn new(schema: SchemaRef) -> Self {
+        let identifier_field_ids = schema.identifier_field_ids().collect::<HashSet<i32>>();
+        Self {
+            case_sensitive: false,
+            schema,
+            identifier_field_ids,
+            deletes: HashSet::new(),
+        }
+    }
+
+    /// Set case sensitivity when updating schema by column names.
+    pub fn set_case_sensitivity(mut self, case_sensitivity: bool) -> Self {
+        self.case_sensitive = case_sensitivity;
+        self
+    }
+
+    /// Deletes a column from a table.
+    ///
+    /// # Arguments
+    ///
+    /// * `column_name` - The path to the column.
+    ///
+    /// # Returns
+    ///
+    /// Returns the `UpdateSchema` with the delete operation staged.
+    pub fn delete_column(&mut self, column_name: Vec<String>) -> Result<&mut Self> {
```

Comment:
One thing I'm hesitant about is, is it better to provide a transaction action `SetSchemaAction`.